Add integration testing and example for skipBuildDependencies option

pscarey opened the pull request:
Fixes #1346 

Currently, it is not possible to deploy a `helm` chart with the following structure:

```
Chart.yaml
values.yaml
requirements.yaml # Declares local dependency charts, no repository field
templates/ # Chart templates
charts/
    subchart/
        Chart.yaml
        values.yaml
        templates/
```

The `helm dep build` call breaks on charts following this structure, despite the chart being valid for deployment with `helm install` and `helm upgrade`.

A key production example of this is the `istio` helm chart, found [here](https://github.com/istio/istio/tree/1.0.4/install/kubernetes/helm/istio).

This pull request fixes this issue with a `skipDependencyBuild` option for `helm` releases (which skips the `helm dep build` phase) and includes documentation, an example (based off the existing helm example) and adds a new test for the validity of the `skipDependencyBuild` option.

balopat commented on integration/examples/helm-deployment-dependencies/skaffold.yaml:
```diff
@@ -0,0 +1,28 @@
+apiVersion: skaffold/v1beta1
```
```suggestion
apiVersion: skaffold/v1beta2
```

nkubala commented:
once you rebase the docs LGTM

pscarey commented:
Looks like I've messed up the rebase, give me a bit to sort.

pscarey commented:
OK, looks better. Will progress #1371 when this is merged.

nkubala commented:
sorry, looks like you need one more rebase 😨 send a ping once you do and we'll get this merged so you don't have to keep fighting with it. also, once you get https://github.com/GoogleContainerTools/skaffold/pull/1371 ping us there as well so we can get that reviewed/merged ASAP

balopat commented on pkg/skaffold/deploy/helm.go:
```diff
@@ -161,7 +161,7 @@ func (h *HelmDeployer) deployRelease(ctx context.Context, out io.Writer, r lates
 	// Dependency builds should be skipped when trying to install a chart
 	// with local dependencies in the chart folder, e.g. the istio helm chart.
 	// This decision is left to the user.
-	if !r.SkipBuildDependencies {
```
we should do renames only if it's really necessary - I feel like the benefit here is marginal - even though I understand where you're coming from. Do you mind reverting this?

balopat commented:
In order to merge this in before next release (next week) we need: 
- [ ] rebase to latest 
- [ ] remove the v1beta4 changes 
- [ ] revert the rename

I'm happy to add this @pscarey  if you are okay with that.

pscarey commented:
Thanks @balopat 

I've updated the branch to the latest master and removed the v1beta4 change, and updated the schemas for tests to pass.

The rename was actually fixing a mistake when I reverted the name to my first draft. The original comment in this PR refers to the option as 'skipDependencyBuild'. In commit 2acb227 I accidentally rolled this back to my first draft of 'skipBuildDependencies', so I corrected it back to 'skipDependencyBuild' in 4e23a49.

Please let me know if there's anything else you require. Keen to start using this shortly at work, pending the other PR. Hopefully that can be simpler.

Eraac commented:
> Please let me know if there's anything else you require. Keen to start using this shortly at work, pending the other PR. Hopefully that can be simpler.

You can already use an option for skip dependencies since #1254 was merged and release with 0.22 (https://github.com/GoogleContainerTools/skaffold/releases/tag/v0.22.0)

pscarey commented:
Ah, that's where the naming confusion has come from. This PR then only includes an integration example over that (+ currently, the re-name). Please let me know if you want the PR re-named + integration example to be merged, or alternatively just close the PR.

R.E. usage, we need #1371, which I can now update off the current master & can be looked at again.

balopat commented:
There is value in the integration example - but please revert the rename - I think that the value there is outweighed by the noise it creates during upgrades.

pscarey commented:
Rename has been reverted. Related PR has been separated out from this to work against the current master.

Only other change made was that to pass tests, the integrations example had to also be added to the root examples folder, contrary to the previous request by @balopat 

> Thank you for the contribution, can you change the examples only under the integration folder? The examples in the root folder are for the latest released version for compatibility reasons.
